internal/store/postgres/model/grant.go

```diff
@@ -162,6 +185,15 @@ func (m Grant) ToDomain() (*domain.Grant, error) {
 	if !m.RevokedAt.IsZero() {
 		grant.RevokedAt = &m.RevokedAt
 	}
+	if m.RestoreReason.Valid {
+		grant.RestoreReason = m.RestoreReason.String
+	}
```

Comment:
can also consider below

```suggestion
	grant.RestoreReason = m.RestoreReason.String
```